apps/aecore/lib/aecore/txs/pool/worker.ex

```diff
@@ -78,4 +115,81 @@ defmodule Aecore.Txs.Pool.Worker do
     {:reply, tx_pool, %{}}
   end
 
+  def add_proof_to_txs(user_txs) do
+    blocks_for_user_txs =
+    for user_tx <- user_txs do
+      get_block_by_txs_hash(user_tx.txs_hash)
+    end
+    merkle_trees =
+    for block <- blocks_for_user_txs do
+      BlockValidation.build_merkle_tree(block.txs)
+    end
+    user_txs_trees = Enum.zip(user_txs, merkle_trees)
+    proof =
+    for user_tx_tree <- user_txs_trees  do
+      {tx, tree} = user_tx_tree
+      key = :erlang.term_to_binary(
+        tx
+        |> Map.delete(:txs_hash)
+        |> Map.delete(:block_hash)
+        |> Map.delete(:block_height)
+        |> Map.delete(:signature)
+        |> TxData.new()
+      )
+      |> TxData.hash_tx()
+      merkle_proof = :gb_merkle_trees.merkle_proof(key, tree)
+      serialized_merkle_proof = serialize_merkle_proof(merkle_proof, [])
+      Map.put_new(tx, :proof, serialized_merkle_proof)
+    end
+  end
+
+  ## Private functions
+
+  defp split_blocks([block | blocks], address, txs) do
+    user_txs = check_address_tx(block.txs, address, txs)
+    case user_txs do
+      [] -> split_blocks(blocks, address, user_txs)
+      _ ->
+        block_user_txs =
+      for block_user_txs <- user_txs do
+        block_user_txs
+        |>  Map.put_new(:txs_hash, block.header.txs_hash)
+        |>  Map.put_new(:block_hash, block.header.chain_state_hash)
+        |>  Map.put_new(:block_height, block.header.height)
+      end
+        split_blocks(blocks, address, block_user_txs)
+    end
```

Comment:
You can return `block_user_txs` for each case and just call the `split_blocks()` once. Also `if do` can replace `case do` here